<?xml version="1.0"?>

<launch>
  <arg name="use_sim_time" default="true" />
  <arg name="gui" default="true" />
  <arg name="headless" default="false" />
  <arg name="world_name" default="$(find jackal_gazebo)/worlds/empty_world.world" />
  <arg name="ns1" default="jackal1" />
  <arg name="ns2" default="jackal2" />
  <arg name="config1_id" default="1" />
  <arg name="config2_id" default="2" />

  <!-- Short-term hack to support the original front_laser:=true argument for spawning
       the simulator. This will be removed in favour of using the config:=x arg instead. -->
  <arg name="front_laser" default="true" />
  <arg name="default_config" value="front_laser" if="$(arg front_laser)" />
  <arg name="default_config" value="base" unless="$(arg front_laser)" />
  <!-- end of hack -->

  <!-- Configuration of Jackal which you would like to simulate.
       See jackal_description for details. -->
  <arg name="config" default="$(arg default_config)" />

  <!-- Optionally enable teleop for the simulation -->
  <arg name="joystick" default="true" />

  <!-- Launch Gazebo with the specified world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="debug" value="1" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="use_sim_time" value="$(arg use_sim_time)" />
    <arg name="headless" value="$(arg headless)" />
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="verbose" value="true"/>
  </include>

  <!-- Spawn Jackal -->
  <group ns="$(arg ns1)">
    <!-- <param name="tf_prefix" value="$(arg ns1)"/> -->
    <include file="$(find jackal_gazebo)/launch/spawn_jackal.launch">
      <arg name="x" value="0" />
      <arg name="y" value="0" />
      <arg name="z" value="0.0" />
      <arg name="ns" value="$(arg ns1)" />
      <arg name="yaw" value="0" />
      <arg name="config" value="$(arg config)" />
      <arg name="config_id" value="$(arg config1_id)"/>
      <arg name="joystick" value="$(arg joystick)" />
    </include>
  </group>
  
  <group ns="$(arg ns2)">
    
    <include file="$(find jackal_gazebo)/launch/spawn_jackal.launch">
      <arg name="x" value="-5.0" />
      <arg name="y" value="0" />
      <arg name="z" value="0.0" />
      <arg name="ns" value="$(arg ns2)" />
      <arg name="yaw" value="0" />
      <arg name="config" value="$(arg config)" />
      <arg name="config_id" value="$(arg config2_id)"/>
      <arg name="joystick" value="$(arg joystick)" />
    </include>

  </group>


  <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_keyboard">
    <remap from= "/cmd_vel" to="/jackal1/jackal_velocity_controller/cmd_vel" />
  </node>

<node pkg="jackal_gazebo" type = "jackal_listener.py" name = "jackal_tf2_listener" output = "screen">
</node>
  
</launch>
